### ISSUE TITLE
Constructing pipes bypasses pipe stacking checks

### ISSUE THREAD

**Issue by Kayek14** [space-wizards/space-station-14#21187]
## Description
Apparently if you build a gas pipe on a preexisting gas pipe they will share gases.

**Reproduction**
Construct a pipe on a preexisting pipe.

**Screenshots**

The grey pipe has plasma in it. Disto is getting plasma in it some how. It is built on the same tile as the distro pipe, but not connected:
![image](https://github.com/space-wizards/space-station-14/assets/106507028/f5f351f8-b7b0-40c0-b6cd-d66f00a937e7)

Problem area here and the bend to the left of the box:
![image](https://github.com/space-wizards/space-station-14/assets/106507028/ca7a7c12-f423-479d-8415-7a3f61ca1eda)

Logs show that the grey pipe was built on top of the distro pipe. I believe the distro pipe is mapped in.
![image](https://github.com/space-wizards/space-station-14/assets/106507028/0a5d00e1-e19f-43cc-b64d-a398e84fc81a)

**Additional context**
Talked about this in discord with notafet and was asked to make this issue.

**Comment by deltanedas** [comment]
yes thats how atmos works colour is just a cosmetic thing

**Comment by Kayek14** [comment]
These two pipes are completly seperate. Pipe layering should not happen. At best it leads to unintended exploits. At worst it leads to an accidental plasmaflood as seen in the picture above.

Its unintuitive and leads to alot of problems

**Comment by deltanedas** [comment]
they arent separate though they join between the bend and straight pipe

**Comment by Kayek14** [comment]
There was a blue pipe under that. The grey pipe was built on top of the blue pipe. Pipes shouldent layer and connect like this

**Comment by deltanedas** [comment]
pipe layering isnt a thing, why shouldnt they connect

**Comment by daerSeebaer** [comment]
Right now, piping works in the way that all pipes that could connect, will connect. So in your setup, you created 3 new connections (blue left - grey right, grey left - blue right, grey left - grey right), which in turn just put them all into the same pipe network.

But yes, this should not work as in you should have never been able to construct it that way in the first place.

Maybe we could do a check when a pipe gets constructed if it violates stacking rules? And if yes it just unanchors and mabye wiggles a bit to make it easily identifiable

**Comment by Arctic-Husky** [comment]
Maybe if there was a way to construct multiple pipes in the same tile like in 13 this wouldn't be a problem

**Comment by daerSeebaer** [comment]
So I did some more investigating and I did not find any mechanic that would prevent pipes from stacking. @Kayek14 are you sure that there are checks that are bypassed as you claim in the issue title?

> Maybe if there was a way to construct multiple pipes in the same tile like in 13 this wouldn't be a problem

It would still be a problem since having two pipes that connect at the same side on the same layer is not desirable. It would only make this problem less likely to occur without solving it.
